### PR TITLE
Refactor game controller

### PR DESCRIPTION
**Pull request by wcstorey**
@dm-istomin @matthewhuff89

### CODE REVIEW

File: app/game/game_engine/controller.rb [hunk 1]
```diff
@@ -1,61 +1,87 @@
...
+
+      case game_state.phase
+      when :setup
+        if phase_time_check(game_state, :setup_time)
           game_state.phase = :move
-          game_state.time = Time.now
           GameEngine::GameResolver.deal_cards(game_state)
-          GameEngine::Cache.save_game_state(game_state)
-          GameEngine::IO.output_player_data(game_state, player_id)
+          gamestate_time_save_output(game_state, player_id)
```
Comment (dm-istomin) [inline app/game/game_engine/controller.rb]: @wcstorey - This name could be more descriptive. How about "save_time_and_output"?

File: app/game/game_engine/controller.rb [hunk 2]
```diff
@@ -1,61 +1,87 @@
...
-          end
+      when :move
+        if phase_time_check(game_state, :move_time) && !selection_made?(current_player)
+            current_player.selection << nil
+            save_game_state(game_state)
+            output_player_data(game_state, player_id)
         end
 
-        if !game_state.player_one.selection.empty? && !game_state.player_two.selection.empty?
+        if selection_made?(player_one) && selection_made?(player_two)
```
Comment (dm-istomin) [inline app/game/game_engine/controller.rb]: @wcstorey - How about making this a method on GameController? Something like "both_players_moved?"

File: app/game/game_engine/controller.rb (hunk 1 above)
Comment (wcstorey) [inline app/game/game_engine/controller.rb]: updatetime_savegamestate_and_outputplayerdata

File: app/game/game_engine/controller.rb (hunk 2 above)
Comment (wcstorey) [inline app/game/game_engine/controller.rb]: bothplayersmoved?(player_one, player_two)

**Comment by dm-istomin** [comment]
@wcstorey - Ok, reviewed changes and merging.

**Pull request merged**